src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -154,13 +153,15 @@ void ZooKeeper::init(ZooKeeperArgs args_, std::unique_ptr<Coordination::IKeeper>
                     LOG_DEBUG(log, "Trying to connect to a more optimal node {}", optimal_host.host);
                     ShuffleHosts node{optimal_host};
                     std::unique_ptr<Coordination::IKeeper> new_impl = std::make_unique<Coordination::ZooKeeper>(node, args, zk_log);
-                    Int8 new_node_idx = new_impl->getConnectedNodeIdx();
 
-                    /// Maybe the node was unavailable when getting AZs first time, update just in case
-                    if (args.availability_zone_autodetect && availability_zones[new_node_idx].empty())
+                    if (auto new_node_idx = new_impl->getConnectedNodeIdx(); new_node_idx)
```

Comment:
It's always true (unless we have a bug, but it's better to add an assert or throw a LOGICAL_ERROR in this case). Moreover, even if this happens somehow, this code will ignore the issue and will finalize the current session without a reason